flang/lib/Lower/SymbolMap.h

```diff
@@ -67,10 +67,18 @@ struct SymbolBox : public fir::details::matcher<SymbolBox> {
   // Generalized derived type variable
   using Derived = fir::BoxValue;
 
+  // Pointer or allocatable variable
   using PointerOrAllocatable = fir::MutableBoxValue;
 
+  // Non pointer/allocatable variable that must be tracked with
+  // a fir.box (either because it is not contiguous, or assumed rank, or assumed
+  // type, or polymorphic, or because the fir.box is describing an optional
+  // value and cannot be read into one of the other category when lowering the
+  // symbol).
+  using IrBox = fir::IrBoxValue;
+
   using VT = std::variant<Intrinsic, FullDim, Char, CharFullDim, Derived,
-                          PointerOrAllocatable, None>;
+                          PointerOrAllocatable, IrBox, None>;
```

Comment:
As before: `MutableBox`?

Reply:
`MutableBoxValue` is in here, but is named `PointerOrAllocatable` in SymbolBox.